Model/Sync/Catalog/Processor.php

```diff
@@ -433,8 +439,31 @@ public function syncItems($collectionItems, $storeId, $isVisibleVariantsSync = f
         }
 
         if ($visibleVariantsDataValues && !$isVisibleVariantsSync) {
-            $this->syncItems($visibleVariantsDataValues, $storeId, true);
+            $wereVisibleVariantsCreationAttemptsSuccessful = $this->syncItems($visibleVariantsDataValues, $storeId, true);
+            if (!$wereVisibleVariantsCreationAttemptsSuccessful) {
+                $this->yotpoCatalogLogger->info(
+                    __(
+                        'API errors occurred while trying to create visible variants - Store ID: %1, Store Name: %2',
+                        $storeId,
+                        $this->coreConfig->getStoreName($storeId)
+                    )
+                );
+
+                return false;
```

Comment:
this is redundant and confusing becuase the function is recusrsive.
lets just assign the value.